### PR TITLE
does apply now oob

### PR DESCRIPTION
**Pull request by wibed**
1. Adjusted indentation to match Helm's template syntax.
2. Corrected the placement of curly braces ({{ }}) to ensure they enclose the template expressions.
3. Removed extra spaces for better readability.
4. Included conditional blocks for bootstrap and backup sections.

### CODE REVIEW

**Comment by itay-grudev** [comment]
@wibed Are you a user or a bot? I am genuinely surprised to see this PR.

**Comment by wibed** [comment]
a user. i am just genuinely grateful for your contribution

File: charts/cluster/templates/cluster.yaml [hunk 1]
```diff
@@ -2,63 +2,69 @@ apiVersion: postgresql.cnpg.io/v1
...
-  {{ include "cluster.bootstrap" . | nindent 2 }}
-  {{ include "cluster.backup" . | nindent 2 }}
+    - name: {{ include "cluster.fullname" . }}-monitoring
+      key: custom-queries
+{{- end }}
+
+{{- if (and (hasKey .Values.cluster "bootstrap") .Values.cluster.bootstrap.enabled) }}
+{{ include "cluster.bootstrap" . | indent 2 }}
+{{- end }}
+
```
Comment (itay-grudev) [inline charts/cluster/templates/cluster.yaml:67]: There is a check inside the template here:
https://github.com/itay-grudev/cloudnative-pg-charts/blob/e54167ff0359e650ea507584438715d84266230f/charts/cluster/templates/_backup.tpl#L3

Is the change here required for some reason?

File: charts/cluster/templates/cluster.yaml [hunk 2]
```diff
@@ -2,63 +2,69 @@ apiVersion: postgresql.cnpg.io/v1
...
+{{- if (and (hasKey .Values.cluster "backup") .Values.cluster.backup.enabled) }}
+{{ include "cluster.backup" . | indent 2 }}
+{{- end }}
```
Comment (itay-grudev) [inline charts/cluster/templates/cluster.yaml:70]: There is a check already here:
https://github.com/itay-grudev/cloudnative-pg-charts/blob/e54167ff0359e650ea507584438715d84266230f/charts/cluster/templates/_backup.tpl#L3

Why was this required?

Comment (wibed) [reply charts/cluster/templates/cluster.yaml:70]: i couldnt run it without it. reporting an error.

File: charts/cluster/templates/cluster.yaml [hunk 3]
```diff
@@ -2,63 +2,69 @@ apiVersion: postgresql.cnpg.io/v1
...
-    {{- end }}
-  {{ include "cluster.bootstrap" . | nindent 2 }}
-  {{ include "cluster.backup" . | nindent 2 }}
+    - name: {{ include "cluster.fullname" . }}-monitoring
+      key: custom-queries
+{{- end }}
+
+{{- if (and (hasKey .Values.cluster "bootstrap") .Values.cluster.bootstrap.enabled) }}
+{{ include "cluster.bootstrap" . | indent 2 }}
+{{- end }}
```
Comment (itay-grudev) [inline charts/cluster/templates/cluster.yaml:66]: I don't think there is such a key: `.Values.cluster.bootstrap`. This will break the bootstrap section.

Comment (itay-grudev) [reply charts/cluster/templates/cluster.yaml:66]: Also the bootstrap section is generated in a more-complicated fashion, dependent on several settings. Take a look here:

https://github.com/itay-grudev/cloudnative-pg-charts/blob/main/charts/cluster/templates/_bootstrap.tpl

File: charts/cluster/templates/cluster.yaml (hunk 2 above)
Comment (itay-grudev) [reply charts/cluster/templates/cluster.yaml:70]: They may be a symptom of another issue. What was the error?

File: charts/cluster/templates/cluster.yaml (hunk 3 above)
Comment (wibed) [reply charts/cluster/templates/cluster.yaml:66]: i did get the same error on bootstrap aswell.
i havent tried one with the bootstrap section populated in any way.

File: charts/cluster/templates/cluster.yaml (hunk 2 above)
Comment (itay-grudev) [reply charts/cluster/templates/cluster.yaml:70]: Also the correct setting should be:

```
backups.enabled: false
```

as opposed to:
```
cluster.backup.enabled: false
```

This chart does things slightly differently compared to the standard CNPG API, focusing  on easy recovery procedures and convention over configuration.

Comment (wibed) [reply charts/cluster/templates/cluster.yaml:70]: it might be that cluster.backup does not exist.
but this line does:
`{{ include "cluster.backup" . | indent 2 }}`

File: charts/cluster/templates/cluster.yaml (hunk 3 above)
Comment (itay-grudev) [reply charts/cluster/templates/cluster.yaml:66]: Can you share your `values.yaml`. There may be something interesting going in that I haven't accounted for?

Comment (wibed) [reply charts/cluster/templates/cluster.yaml:66]: of course here is the minimal helm chart:

```yaml
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization
namespace: cnpg-system

helmGlobals:
  chartHome: ../../../../base/cnpgdb


helmCharts:
- name: cluster
  releaseName: cluster0
  namespace: cnpg-system
  includeCRDs: true
  valuesInline:
    cluster: 
      priorityClassName: high-priority
      storage:
        storageClass: "csi-driver-nfs"
  

patches:
  - path: values.yaml
    target:
      kind: HelmRelease

````

Comment (itay-grudev) [reply charts/cluster/templates/cluster.yaml:66]: I found it. Thanks. It's an issue in the bootstrap section.

Comment (wibed) [reply charts/cluster/templates/cluster.yaml:66]: sincerely, it might be a documentation issue. 

could you elaborate on the bootstrap section a bit?
sadly the links are dead

Comment (itay-grudev) [reply charts/cluster/templates/cluster.yaml:66]: I fixed the issue and merged the fix into the main branch. The changes for backups and bootstrap section should no longer be required.

Comment (itay-grudev) [reply charts/cluster/templates/cluster.yaml:66]: As to the bootstrap section, this chart is designed to run in several modes:
* `standalone` - Creates new or updates an existing CNPG cluster. This is the default mode.
* `replica` - Creates a replica cluster from an existing CNPG cluster. **_Note_ that this mode is not yet supported.**
* `recovery` - Recovers a CNPG cluster from a backup, object store or via pg_basebackup.

The idea is during disaster recovery to get your original values, set the mode to `recovery`, swap the backup and bootstrap sections and run it. See [Recovery](https://github.com/itay-grudev/cloudnative-pg-charts/blob/main/charts/cluster/docs/Recovery.md). The simpler the procedure - the faster you can get it done and restore normal operations.
